### PR TITLE
Make sure release commit signed and tagged

### PR DESCRIPTION
**Pull request by legleux**
Best practices, good housekeeping and help me look stuff up.

Looks like the annotated tag train has derailed. This is an attempt to enforce it.

### CODE REVIEW

File: .githooks/pre-push [hunk 1]
```diff
@@ -1,3 +1,58 @@
...
+
+# git for-each-ref refs/tags  # see which tags are annotated and which are lightweight. Annotated tags are "tag" objects.
+# # Set these so your commits and tags are always signed
+# git config commit.gpgsign true
+# git config tag.gpgsign true
+
+verify_commit_signed() {
+    if git verify-commit HEAD &> /dev/null; then
+        :
+        # echo "HEAD commit seems signed..."
```
Comment (cindyyan317) [inline .githooks/pre-push:12]: do you want to remove the comments

Comment (kuznetsss) [reply .githooks/pre-push:12]: It looks like a useful comment.

File: .githooks/pre-push [hunk 2]
```diff
@@ -1,3 +1,58 @@
...
+        # echo "Tag appears signed"
+        return 0
+    else
+        echo "$version tag isn't signed"
+        echo "Sign it with [git tag -ams\"$version\" $version]"
+        exit 1
+    fi
+}
+
+while read local_ref local_oid remote_ref remote_oid; do
```
Comment (kuznetsss) [inline .githooks/pre-push:42]: Is it predefined variables in git hooks?

File: .githooks/pre-push (hunk 1 above)
Comment (legleux) [reply .githooks/pre-push:12]: my comments disappeared?
I had said I was on the fence and usually don't leave comments like this but I kept uncommenting while debugging so I decided to leave them in. :shrug:

File: .githooks/pre-push (hunk 2 above)
Comment (legleux) [reply .githooks/pre-push:42]: Sort of? (from the sample hook)
```
# This hook is called with the following parameters:
#
# $1 -- Name of the remote to which the push is being done
# $2 -- URL to which the push is being done
#
# If pushing without using a named remote those arguments will be equal.
#
# Information about the commits which are being pushed is supplied as lines to
# the standard input in the form:
#
#   <local ref> <local sha1> <remote ref> <remote sha1>
```

**Review by kuznetsss — APPROVED** [review]

**Pull request merged**